src/data/subtype.lean

```diff
@@ -3,14 +3,26 @@ Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl
 -/
-import tactic.lint
 import tactic.ext
+import tactic.lint
 import tactic.simps
 
+/-!
+# Subtypes
+
+This file provides basic API for subtypes, which are defined in core.
+
+A subtype is a type made from restricting another type, say `α`, to its elements that satisfy some
+predicate, say `p : α → Prop`. Specifically, it is the type of pairs `⟨val, property⟩` where
+`val : α` and `property : p val`. A subtype has a natural coercion to the parent type, by coercing
+`⟨val, property⟩` to `val`. Subtypes can be thought of as bundled sets, the difference being that
+elements of a set are still of type `α` while elements of a subtype aren't.
+-/
```

Comment:
Would it make sense to mention the notation here?

Reply:
Oh yeah, dumb me!